### PR TITLE
Add ApplicativeError for EitherT documentation

### LINKED ISSUE #3177: EitherT.fromMonadError
I think #1644 is related.

Basicly something like:
```
import cats.implicits._
import cats.data.EitherT
import cats.MonadError

def fromMonadError[F[_], A, B](me: F[B])(implicit ev: MonadError[F, A]) = {
  EitherT(me.map(Either.right[A, B]).recover{case ex => Either.left[A, B](ex)})
}
```

Or if we have this functionality already maybe it would worth a mention in the EitherT/MonadError documentation.

If others think this is useful, I try to PR it :)

### PR DESCRIPTION
**Pull request by tg44**
fixes #3177

### CODE REVIEW

File: docs/src/main/tut/datatypes/eithert.md [hunk 1]
```diff
@@ -145,6 +145,19 @@ val myOptionET = EitherT.fromOption[Future](myOption, "option not defined")
...
+
+An `ApplicativeError[F, E]` or a `MonadError[F, E]`, can be converted into `EitherT[F, E, A]` 
+using the `attemptT` method on them.
+
+```tut:book
+val myTry: Try[Int] = Try(2)
+val myFuture: Future[String] = Future.failed(new Exception())
+
+val myTryET = myTry.attemptT
+val myFutureET = myFuture.attemptT
```
Comment (LukaJCB) [inline docs/src/main/tut/datatypes/eithert.md]: Should we add type annotations to these as well to make it more obvious what the function does? :)

Comment (tg44) [reply docs/src/main/tut/datatypes/eithert.md]: I had a dilemma with this, but the tut:book creates a kinda great output so I let it go...
Probably using tut:silent and type annotations would be an another good option.

Comment (tg44) [reply docs/src/main/tut/datatypes/eithert.md]: So what do you think? Should I add the return types, or let the `tut:book` to do its magic?

Comment (djspiewak) [reply docs/src/main/tut/datatypes/eithert.md]: IMO I think it's better to add the type annotations, just to be fully clear even in the original form. Also at some point I'm assuming we'll be switching from tut to mdoc, so it's worth thinking a bit ahead.

**Comment by tg44** [comment]
fixed it

**Comment by solarmosaic-kflorence** [comment]
@LukaJCB This is very helpful documentation, but it does not appear on the cats website: https://typelevel.org/cats/datatypes/eithert.html

See related stackoverflow question: https://stackoverflow.com/questions/54931204/recovering-underlying-future-into-cats-eitherts-left

**Pull request merged**